File: sample_code.m
clear; close all;

img1 = imread('helmet_mkv_0000000000.jpg');
img2 = imread('helmet_mkv_0000000004.jpg');
Frame1 = rgb2gray(img1);
Frame2 = rgb2gray(img2);

FFT1=fft2(Frame1);
FFT2=fft2(Frame2);
Amp1=abs(FFT1);
Amp2=abs(FFT2);
Phase1=angle(FFT1);
Phase2=angle(FFT2);
mMap1=abs(ifft2((Amp2-Amp1).*exp(i*Phase1)));
mMap2=abs(ifft2((Amp2-Amp1).*exp(i*Phase2)));
mMap=mat2gray(mMap1.*mMap2);

figure, imshow(img1), title('image1');
figure, imshow(img2), title('image2');

figure, imshow(mat2gray(mMap1)), title('saliency map1');
figure, imshow(mat2gray(mMap2)), title('saliency map2');
figure, imshow(mMap), title('final saliency map');

File: sample_code.py
from skimage import color

import matplotlib.pyplot as plt
import numpy as np
import cv2

img1 = plt.imread('helmet_mkv_0000000000.jpg')
img2 = plt.imread('helmet_mkv_0000000004.jpg')
frame1 = color.rgb2gray(img1)
frame2 = color.rgb2gray(img2)

fft1 = np.fft.fft2(frame1)
fft2 = np.fft.fft2(frame2)
amp1 = np.abs(fft1)
amp2 = np.abs(fft2)
phase1 = np.angle(fft1)
phase2 = np.angle(fft2)

m_map1 = np.abs(np.fft.ifft2((amp2-amp1) * np.exp(1j * phase1)));
m_map2 = np.abs(np.fft.ifft2((amp2-amp1) * np.exp(1j * phase2)));
m_map = m_map1 * m_map2


fig, ax = plt.subplots(2, 2)
ax[0, 0].imshow(img1)
ax[0, 1].imshow(img2)
ax[1, 0].imshow(m_map1)
ax[1, 1].imshow(m_map2)

fig, ax = plt.subplots()
im = ax.imshow(m_map, cmap='gray')
fig.colorbar(im)
plt.show()